apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-plugin/src/main/java/co/elastic/apm/agent/opentelemetry/sdk/OTelSpanBuilder.java

```diff
@@ -182,6 +188,11 @@ public Span startSpan() {
         // user outcome to provide higher priority and avoid inferring outcome from any reported exception
         span.withUserOutcome(Outcome.UNKNOWN);
 
+        // Add the links to the span
+        for (int i = 0; i < links.size(); i++) {
+            span.addSpanLink(TraceContext.fromParentContext(), links.get(i));
```

Comment:
We can delay the use of `((OTelSpanContext) spanContext).getElasticTraceContext()` until here.
I think it better reflects who's responsible to what - the user is responsible to make sure that at this point all OTel states are correct and it is our responsibility to take it from here.

Reply:
agreed